cpp/src/ice2slice/Gen.cpp

```diff
@@ -43,13 +74,26 @@ namespace
 
         if (auto metadata = p->getMetadataArgs("cs:namespace"))
         {
-            hasCSharpNamespaceAttribute = true;
-            return *metadata + "." + csharpNamespace;
+            return *metadata;
+        }
+        else
+        {
+            return "";
+        }
```

Comment:
There's a nice convenience function for this:
```
return getMetadataArgs("cs:namespace").value_or("");
```
Unfortunately, GitHub won't let me suggest it, since the `if` wasn't changed in this PR...